packages/teleterm/src/ui/services/workspacesService/accessRequestsService.ts

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { ResourceKind } from 'e-teleterm/ui/DocumentAccessRequests/NewRequest/useNewRequest';
+import { AccessRequest } from 'e-teleport/services/workflow';
+
+import { PendingAccessRequest } from '../workspacesService';
+
+export class AccessRequestsService {
+  constructor(
+    private getState: () => {
+      isAccessRequestsBarCollapsed: boolean;
+      pendingAccessRequest: PendingAccessRequest;
+      assumed: Record<string, AccessRequest>;
+    },
+    private setState: (
+      draftState: (draft: {
+        isAccessRequestsBarCollapsed: boolean;
+        pendingAccessRequest: PendingAccessRequest;
+        assumed: Record<string, AccessRequest>;
```

Comment:
I thought about this same thing tbh. I believe its because when adding a resource, we must check the type + name since the name isn't guaranteed to be unique (imagine a node named `teleport` and a db named `teleport`), however, I am storing by the ID anyway and I'm not sure it has the same issue? Sort of a relic of the past but, either either Object.keys in places, or array.filter/map in others to get counts+types in other places.

Reply:
So the key has structure `type-resourceName`?